lib/src/ruby/module.cc

```diff
@@ -304,13 +298,11 @@ namespace facter { namespace ruby {
 
     void module::clear_facts(bool clear_collection)
     {
-        auto ruby = api::instance();
+        auto const& ruby = api::instance();
```

Comment:
This is not a static method, so I don't see any way an exception would be thrown here. So skipping the try/catch seems safe.